openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,464 @@
+        print(config_global["filepath"])
+        config_data = _get_config_data(
+            config_global["filepath"], anatomy_data
+        )
+
+    if not config_data:
+        raise FileExistsError(
+            "No OCIO config found in settings. It is "
+            "either missing or there is typo in path inputs"
+        )
+
+    return config_data
+
+
+def _get_config_data(path_list, anatomy_data):
+    """Path formating in interation
+
+    Args:
+        path_list (list[str]): list of abs paths
+        anatomy_data (dict): formating data
+
+    Returns:
+        dict: config data
+    """
+    # first try host config paths
+    for path_ in path_list:
+        formated_path = _format_path(path_, anatomy_data)
+        if not os.path.exists(formated_path):
+            continue
+
+        return {
+            "path": os.path.normpath(formated_path),
+            "template": path_
+        }
+
+
+def _format_path(tempate_path, anatomy_data):
+    """Single template path formating.
+
+    Args:
+        tempate_path (str): template string
+        anatomy_data (dict): formating data
+
+    Returns:
+        str: absolute formated path
+    """
+    formatting_data = deepcopy(anatomy_data)
+
+    # format the path for potential env vars
+    formatting_data.update(dict(**os.environ))
+
+    # format path for anatomy keys
+    formatted_path = StringTemplate(tempate_path).format(
+        formatting_data)
+
+    return os.path.abspath(formatted_path)
+
+
+def get_imageio_file_rules(project_name, host_name, project_settings=None):
+    """Get ImageIO File rules from project settings
+
+    Args:
+        project_name (str): project name
+        host_name (str): host name
+        project_settings (dict, optional): project settings.
+                                           Defaults to None.
+
+    Returns:
+        dict: file rules data
+    """
+    project_settings = project_settings or get_project_settings(project_name)
+
+    imageio_global, imageio_host = _get_imageio_settings(
+        project_settings, host_name)
+
+    # get file rules from global and host_name
+    frules_global = imageio_global["file_rules"]
+    frules_host = imageio_host["file_rules"]
+
+    # compile file rules dictionary
+    file_rules = {}
+    if frules_global["enabled"]:
+        file_rules.update(frules_global["rules"])
+    if frules_host["enabled"]:
+        file_rules.update(frules_host["rules"])
+
+    return file_rules
+
+
+def _get_imageio_settings(project_settings, host_name):
+    """Get ImageIO settings for global and host
+
+    Args:
+        project_settings (dict): project settings.
+                                           Defaults to None.
+        host_name (str): host name
+
+    Returns:
+        tuple[dict, dict]: image io settings for global and host
+    """
+    # get image io from global and host_name
+    imageio_global = project_settings["global"]["imageio"]
+    imageio_host = project_settings[host_name]["imageio"]
```

Comment:
I feel like this helper method might be better of 'safe-guarding' a host maybe not having implement `imagio` settings yet and do: `project_settings.get(host_name, {}).get("imageio", {})`

```suggestion
    imageio_host = project_settings.get(host_name, {}).get("imageio", {})
```